src/Cli/ConfigGenerator.cs

```diff
@@ -535,6 +541,98 @@ public static bool TryConfigureSettings(ConfigureOptions options, FileSystemRunt
             return WriteRuntimeConfigToFile(runtimeConfigFile, runtimeConfig, fileSystem);
         }
 
+        /// <summary>
+        /// Configures the data source options for the runtimeconfig based on the provided options.
+        /// This method updates the database type, connection string, and other database-specific options in the config file.
+        /// It validates the provided database type and ensures that options specific to certain database types are correctly applied.
+        /// If any validation fails, it logs an error and returns false.
+        /// </summary>
+        /// <param name="options">The configuration options provided by the user.</param>
+        /// <param name="runtimeConfig">The runtime configuration to be updated. This parameter is passed by reference and must not be null if the method returns true.</param>
+        /// <returns>
+        /// True if the data source options were successfully configured and the runtime configuration was updated; otherwise, false.
+        /// </returns>
+        private static bool TryConfigureDataSourceOptions(
+            ConfigureOptions options,
+            [NotNullWhen(true)] ref RuntimeConfig runtimeConfig)
+        {
+            DatabaseType dbType = runtimeConfig.DataSource.DatabaseType;
+            string dataSourceConnectionString = runtimeConfig.DataSource.ConnectionString;
+
+            if (options.DataSourceDatabaseType is not null)
+            {
+                if (!Enum.TryParse(options.DataSourceDatabaseType, ignoreCase: true, out dbType))
+                {
+                    _logger.LogError(EnumExtensions.GenerateMessageForInvalidInput<DatabaseType>(options.DataSourceDatabaseType));
+                    return false;
+                }
+            }
+
+            if (options.DataSourceConnectionString is not null)
+            {
+                dataSourceConnectionString = options.DataSourceConnectionString;
+            }
+
+            Dictionary<string, object?>? dbOptions = new();
+            HyphenatedNamingPolicy namingPolicy = new();
+
+            if (DatabaseType.CosmosDB_NoSQL.Equals(dbType))
+            {
+                AddCosmosDbOptions(dbOptions, options, namingPolicy);
+            }
+            else if (!string.IsNullOrWhiteSpace(options.DataSourceOptionsDatabase)
+                    || !string.IsNullOrWhiteSpace(options.DataSourceOptionsContainer)
+                    || !string.IsNullOrWhiteSpace(options.DataSourceOptionsSchema))
+            {
+                _logger.LogError("Database, Container, and Schema options are only applicable for CosmosDB_NoSQL database type.");
+                return false;
+            }
```

Comment:
when changing dbtype from sql to cosmos, users will need to change and provid cosmos options in a single command, right? Maybe i haven't read far enough but it seems like the 2 step command approach is prevented by this condition.

Reply:
No, it won't be. Because if the user first changes the database, that change is persisted. so the dbType will be cosmosDB when the second command will be run.